File: app/src/main/java/ai/elimu/content_provider/room/dao/ImageDao.java
package ai.elimu.content_provider.room.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ai.elimu.content_provider.room.entity.Image;

@Dao
public interface ImageDao {

    @Insert
    void insert(Image image);

    @Query("SELECT * FROM Image WHERE id = :id")
    Image load(Long id);

    @Query("SELECT * FROM Image WHERE id = :id")
    Cursor loadAsCursor(Long id);

    @Query("SELECT * FROM Image")
    List<Image> loadAll();

    @Query("SELECT * FROM Image")
    Cursor loadAllAsCursor();

    @Query("SELECT * FROM Image i WHERE i.id IN (SELECT Image_id FROM Image_Word WHERE words_id = :wordId)")
    Cursor loadAllByWordLabelAsCursor(Long wordId);

    @Update
    void update(Image image);
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/ImageDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.Image
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ImageDao {
    @Insert
    fun insert(image: Image)

    @Query("SELECT * FROM Image WHERE id = :id")
    fun load(id: Long?): Image?

    @Query("SELECT * FROM Image WHERE id = :id")
    fun loadAsCursor(id: Long?): Cursor

    @Query("SELECT * FROM Image")
    fun loadAll(): MutableList<Image>

    @Query("SELECT * FROM Image")
    fun loadAllAsCursor(): Cursor

    @Query("SELECT * FROM Image i WHERE i.id IN (SELECT Image_id FROM Image_Word WHERE words_id = :wordId)")
    fun loadAllByWordLabelAsCursor(wordId: Long?): Cursor

    @Update
    fun update(image: Image)

    @Query("DELETE FROM Image")
    fun deleteAll()
}
